### PR TITLE
fix Index with name namespace does not exist

### PR DESCRIPTION
**Pull request by xiaoanyunfei**
#### What type of PR is this?
/kind bug
/kind cleanup

#### What this PR does / why we need it:
`podLister.Pods(pod.Namespace).List(labels.Everything())` will get warning logs like 
`listers.go:63] can not retrieve list of objects using index : Index with name namespace does not exist`
when using `coscheduling` plugin

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by kerthcet — COMMENTED** [review]
I think this is a trade-off, it will increase the memory burden especially for a large cluster, but with efficient searching. Do we have a lot of scenarios to search the whole pods inside a namespace? cc @Huang-Wei  have any concern here?

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -499,7 +499,7 @@ func newPodInformer(cs clientset.Interface, resyncPeriod time.Duration) cache.Sh
 	tweakListOptions := func(options *metav1.ListOptions) {
 		options.FieldSelector = selector
 	}
-	informer := coreinformers.NewFilteredPodInformer(cs, metav1.NamespaceAll, resyncPeriod, cache.Indexers{}, tweakListOptions)
+	informer := coreinformers.NewFilteredPodInformer(cs, metav1.NamespaceAll, resyncPeriod, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc}, tweakListOptions)
```
Comment (Huang-Wei) [inline pkg/scheduler/scheduler.go:502]: with #110663, we should be able to customize this indexer on plugin side. @xiaoanyunfei would you want to submit a PR on coscheduling plugin instead?

Comment (kerthcet) [reply pkg/scheduler/scheduler.go:502]: Let me help with this, I'm working on coscheduling plugin. see https://github.com/kubernetes-sigs/scheduler-plugins/pull/654

**Comment by Huang-Wei** [comment]
/close
as this can be fixed in downstream integrators.

**Pull request closed without merging**